test/tests.c

```diff
@@ -3,44 +3,54 @@
 char *folder;
```

Comment:
In my opinion, this file would be easier to read if the functions were grouped by usage. So rather than grouping all the `check_` functions together, then grouping all the `test_` functions together, maybe we could group them by their relatedness to each other. For example:

```
fail
pass

check_create_bucket
check_get_buckets
check_get_bucket
check_get_bucket_id

check_store_file_progress
check_store_file
test_upload

check_store_file_progress_cancel
check_store_file_cancel
test_upload_cancel

check_resolve_file_progress
check_resolve_file
test_download

check_resolve_file_progress_cancel
check_resolve_file_cancel
test_download_cancel

create_test_upload_file
reset_test_upload
reset_test_download

test_api
main
```

Just an idea, and if you think it makes sense more the way it is, feel free to disregard.

Reply:
Agreed, the current organization was inherited and it didn't make sense to move a bunch of stuff around until after all the deletions. I just never got to it. Will do.

If anyone else has to work on this is the future, I'm confident that they'll appreciate it.